.github/workflows/kbs-client-build-and-push.yaml

```diff
@@ -0,0 +1,39 @@
+name: Build and push kbs-client
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build_and_push:
+    env:
+      RUSTC_VERSION: 1.76.0
+    runs-on: ubuntu-22.04
```

Comment:
Hi @portersrc ! I don't remember the last time I built static executables on Ubuntu, so wondering if you need to install the static libc package. On my fedora 39 laptop I had to install glibc-static to test this static build.

Reply:
I don't think I checked this; great question; will check it now.

Reply:
I can confirm that `gcc -static` works on a [simple test](https://github.com/portersrc/infra/actions/runs/8327016995/job/22783806866#step:8:8). Also 22.04 package list includes [libc6-dev](https://packages.ubuntu.com/jammy/allpackages?format=txt.gz), and the (debian) [file list for libc6-dev](https://packages.debian.org/sid/amd64/libc6-dev/filelist) includes libc.a, etc. (So I think we're OK?)

The easier thing is to just verify that this workflow succeeds before merging, but I don't see this workflow getting kicked... Do you know how/why?

Reply:
Thanks for the pointers! I looked at the yaml file, but to test at least part of this PR, I did something cruder: I changed to `on: pull_request` and commented out the ghcr login and push. This at least checks that `make cli-static-x86_64-linux` completes on the runner. It built fine (GHA result [here](https://github.com/confidential-containers/trustee/actions/runs/8473999540/job/23219496678?pr=349)). I changed it back to `on: push`, as intended for this PR. But I guess we need it to merge in order to verify the oras step works (though I tested with a personal repo, and it looks OK).